Add TranslateContent tracking with TranslateRequest

Zelak312 opened the pull request:
This PR aims to make translations started with the api/Translate/content (Used by Bazarr integration) be tracked with TranslateRequest the same way that is normally done when Lingarr does translation by itself

Zelak312 commented:
Add the moment, I made a quick working draft that does no do anything with the Radarr/Sonarr ID

There is a couple points I want to discuss

1. I moved the translation from the controller to the TranslationRequestService
I don't think this should be there really but I was not too sure where this could go.

2. I did not make a Job out of this translation part since currently Bazarr's integration waits for the translated lines to be sent back. This is why I did a TranslateContentAsync function.
I was wondering what you think about it?

3. I Changed the api/Translate/content api to use a different body. That may not be ideal for backwards compatibility. I was wondering if maybe we should add versioning or make it a completely different endpoint

Let me know of any code/plan changes you want me to do!

rowanfuchs commented:
Thank you very much for all the changes and for giving me the option to discuss them. That said, I must say I agree with all your proposals.

1. Great move, I think it belongs there indeed.
2. No job needs to be created, and I realize a new translation request might trigger a job as well, so we’ll need to look into. handling a translation request in a completed state, perhaps?
3. A v2 sounds good to me.

I believe what you’ve created so far is of high quality as well. Thank you for helping me and the Lingarr community.

Zelak312 commented:
Pushed forced since I saw that the PR was on a commit from main that was 2 week old (probably would have been better to merge instead of rebase)... At least now it's up to date with main

Couple new things

1. Progression and working frontend updates should be all done now
2. Added translationRequest migration for SQLite (I noticed I just had done MySQL)
3. I also fixed the cancellation and it is now fully working
A quick note for the cancellation, since they are async translations and not jobs. I had to manually
retain a CancellationTokenSource that I create when the translation start in a static Dictionary of the TranslationRequestService. When a cancellation comes in, if the TranslationRequest doesn't have an JobId it will check in this dictionary to cancel the job if found
